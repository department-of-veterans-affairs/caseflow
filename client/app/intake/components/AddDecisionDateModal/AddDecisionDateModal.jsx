import React, { useMemo, useState } from 'react';
import { useDispatch } from 'react-redux';
import { css } from 'glamor';
import PropTypes from 'prop-types';
import DateSelector from 'app/components/DateSelector';
import Modal from 'app/components/Modal';
import { addDecisionDate } from 'app/intake/actions/addIssues';
import { validateDateNotInFuture } from 'app/intake/util/issues';
import BENEFIT_TYPES from 'constants/BENEFIT_TYPES';

const dateInputStyling = css({
  paddingTop: '24px'
});

const labelStyling = css({
  marginRight: '4px',
});

const AddDecisionDateModal = ({ closeHandler, currentIssue, index }) => {
  const [decisionDate, setDecisionDate] = useState(currentIssue.editedDecisionDate);
  const dispatch = useDispatch();

  // We should disable the save button if there has been no date selected
  // or if the date is in the future
  const isSaveDisabled = useMemo(() => {
    if (!decisionDate) {
      return true;
    }

    return !validateDateNotInFuture(decisionDate);
  }, [decisionDate]);

  const handleOnSubmit = () => {
    dispatch(addDecisionDate({ decisionDate, index }));
  };

  return (
    <div>
      <Modal
        buttons={[
          {
            classNames: ['cf-modal-link', 'cf-btn-link', 'close-modal'],
            name: 'Cancel',
            onClick: closeHandler
          },
          {
            classNames: ['usa-button-blue', 'save-issue'],
            disabled: isSaveDisabled,
            name: 'Save',
            onClick: () => {
              closeHandler();
              handleOnSubmit();
            }
          }
        ]}
        visible
        closeHandler={closeHandler}
        title={currentIssue.editedDecisionDate ? 'Edit Decision Date' : 'Add Decision Date'}
      >
        <div>
          <strong {...labelStyling}>
            Issue:
          </strong>
          {currentIssue.category}
        </div>
        <div>
          <strong {...labelStyling}>
            Benefit type:
          </strong>
          {BENEFIT_TYPES[currentIssue.benefitType]}
        </div>
        <div>
          <strong {...labelStyling}>
            Issue description:
          </strong>
          {currentIssue.nonRatingIssueDescription || currentIssue.description}
        </div>
        <div {...dateInputStyling}>
          <DateSelector
            label="Decision date"
            name="decision-date"
            noFutureDates
            onChange={(value) => setDecisionDate(value)}
            type="date"
            value={decisionDate}
          />
        </div>
      </Modal>
    </div>
  );
};

AddDecisionDateModal.propTypes = {
  closeHandler: PropTypes.func,
  currentIssue: PropTypes.object,
  index: PropTypes.number
};

export default AddDecisionDateModal;
