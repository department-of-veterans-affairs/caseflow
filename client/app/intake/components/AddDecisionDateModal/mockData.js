const closeHandler = () => {
  // eslint-disable-next-line no-console
  console.log('Close');
};

const currentIssue = {
  id: '4310',
  benefitType: 'vha',
  description: 'Beneficiary Travel - Issue Description',
  decisionDate: null,
  decisionReviewTitle: 'Higher-Level Review',
  contentionText: 'Beneficiary Travel - Issue Description',
  category: 'Beneficiary Travel',
};

const index = 0;

export default {
  closeHandler,
  currentIssue,
  index
};
