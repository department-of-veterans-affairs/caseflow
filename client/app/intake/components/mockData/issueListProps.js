export const mockedIssueListProps = {
  editPage: true,
  intakeData: {
    claimant: '358808523',
    claimantType: 'veteran',
    claimantName: 'Bob Smithkeebler',
    veteranIsNotClaimant: false,
    processedInCaseflow: true,
    legacyOptInApproved: false,
    legacyAppeals: [],
    ratings: null,
    editIssuesUrl: '/higher_level_reviews/6545833b-1a6c-4966-823f-7d0037aa5f6a/edit',
    processedAt: '2023-07-28T14:34:36.571-04:00',
    veteranInvalidFields: {
      veteranMissingFields: '',
      veteranAddressTooLong: false,
      veteranZipCodeInvalid: false,
      veteranPayGradeInvalid: false
    },
    requestIssues: [
      {
        id: 6292,
        rating_issue_reference_id: null,
        rating_issue_profile_date: null,
        rating_decision_reference_id: null,
        description: 'Other - stuff and things',
        contention_text: 'Other - stuff and things',
        approx_decision_date: null,
        category: 'Other',
        notes: null,
        is_unidentified: null,
        ramp_claim_id: null,
        vacols_id: null,
        vacols_sequence_id: null,
        ineligible_reason: null,
        ineligible_due_to_id: null,
        decision_review_title: 'Higher-Level Review',
        title_of_active_review: null,
        contested_decision_issue_id: null,
        withdrawal_date: null,
        contested_issue_description: null,
        end_product_code: null,
        end_product_establishment_code: null,
        verified_unidentified_issue: null,
        editable: true,
        exam_requested: null,
        vacols_issue: null,
        end_product_cleared: null,
        benefit_type: 'vha',
        is_predocket_needed: null
      }
    ],
    decisionIssues: [],
    activeNonratingRequestIssues: [
      {
        id: '6291',
        benefitType: 'vha',
        decisionIssueId: null,
        description: 'Caregiver | Eligibility - pact act testing',
        decisionDate: '2023-07-19',
        ineligibleReason: null,
        ineligibleDueToId: null,
        decisionReviewTitle: 'Higher-Level Review',
        contentionText: 'Caregiver | Eligibility - pact act testing',
        vacolsId: null,
        vacolsSequenceId: null,
        vacolsIssue: null,
        endProductCleared: null,
        endProductCode: null,
        withdrawalDate: null,
        editable: true,
        examRequested: null,
        isUnidentified: null,
        notes: null,
        category: 'Caregiver | Eligibility',
        index: null,
        isRating: false,
        ratingIssueReferenceId: null,
        ratingDecisionReferenceId: null,
        ratingIssueProfileDate: null,
        approxDecisionDate: '2023-07-19',
        titleOfActiveReview: null,
        rampClaimId: null,
        verifiedUnidentifiedIssue: null,
        isPreDocketNeeded: null
      }
    ],
    contestableIssuesByDate: [],
    intakeUser: 'SUPERUSER',
    relationships: [
      {
        value: 'CLAIMANT_WITH_PVA_AS_VSO',
        fullName: 'Bob Vance',
        relationshipType: 'Spouse',
        displayText: 'Bob Vance, Spouse',
        defaultPayeeCode: '10'
      },
      {
        value: '1129318238',
        fullName: 'Cathy Smith',
        relationshipType: 'Child',
        displayText: 'Cathy Smith, Child',
        defaultPayeeCode: '11'
      },
      {
        value: 'no-such-pid',
        fullName: 'Tom Brady',
        relationshipType: 'Child',
        displayText: 'Tom Brady, Child',
        defaultPayeeCode: '11'
      }
    ],
    veteranValid: true,
    receiptDate: '2023/07/14',
    veteran: {
      name: 'Bob Smithkeebler',
      fileNumber: '000100009',
      formName: 'Smithkeebler, Bob',
      ssn: '303940217'
    },
    powerOfAttorneyName: 'Clarence Darrow',
    claimantRelationship: 'Veteran',
    asyncJobUrl: '/asyncable_jobs/HigherLevelReview/jobs/385',
    benefitType: 'vha',
    payeeCode: null,
    hasClearedRatingEp: false,
    hasClearedNonratingEp: false,
    informalConference: false,
    sameOffice: null,
    formType: 'higher_level_review',
    contestableIssues: {},
    claimId: '6545833b-1a6c-4966-823f-7d0037aa5f6a',
    featureToggles: {
      useAmaActivationDate: true,
      correctClaimReviews: true,
      covidTimelinessExemption: true
    },
    userCanWithdrawIssues: false,
    addIssuesModalVisible: false,
    nonRatingRequestIssueModalVisible: false,
    unidentifiedIssuesModalVisible: false,
    addedIssues: [
      {
        id: '6291',
        benefitType: 'vha',
        decisionIssueId: null,
        description: 'Other - Other description',
        decisionDate: null,
        ineligibleReason: null,
        ineligibleDueToId: null,
        decisionReviewTitle: 'Higher-Level Review',
        contentionText: 'Other - Other description',
        vacolsId: null,
        vacolsSequenceId: null,
        vacolsIssue: null,
        endProductCleared: null,
        endProductCode: null,
        withdrawalDate: null,
        editable: true,
        examRequested: null,
        isUnidentified: null,
        notes: null,
        category: 'Other',
        index: null,
        isRating: false,
        ratingIssueReferenceId: null,
        ratingDecisionReferenceId: null,
        ratingIssueProfileDate: null,
        approxDecisionDate: null,
        titleOfActiveReview: null,
        rampClaimId: null,
        verifiedUnidentifiedIssue: null,
        isPreDocketNeeded: null
      },
      {
        benefitType: 'vha',
        category: 'Beneficiary Travel',
        description: 'vha camo testing',
        decisionDate: '2023-07-19',
        ineligibleDueToId: null,
        ineligibleReason: null,
        decisionReviewTitle: null,
        isRating: false,
        isPreDocketNeeded: null,
        timely: true,
        editable: true
      }
    ],
    originalIssues: [
      {
        id: '6292',
        benefitType: 'vha',
        decisionIssueId: null,
        description: 'Other - stuff and things',
        decisionDate: null,
        ineligibleReason: null,
        ineligibleDueToId: null,
        decisionReviewTitle: 'Higher-Level Review',
        contentionText: 'Other - stuff and things',
        vacolsId: null,
        vacolsSequenceId: null,
        vacolsIssue: null,
        endProductCleared: null,
        endProductCode: null,
        withdrawalDate: null,
        editable: true,
        examRequested: null,
        isUnidentified: null,
        notes: null,
        category: 'Other',
        index: null,
        isRating: false,
        ratingIssueReferenceId: null,
        ratingDecisionReferenceId: null,
        ratingIssueProfileDate: null,
        approxDecisionDate: null,
        titleOfActiveReview: null,
        rampClaimId: null,
        verifiedUnidentifiedIssue: null,
        isPreDocketNeeded: null
      }
    ],
    requestStatus: {
      requestIssuesUpdate: 'NOT_STARTED'
    },
    requestIssuesUpdateErrorCode: null,
    afterIssues: null,
    beforeIssues: null,
    updatedIssues: null,
    editEpUpdateError: null,
    issueCount: 2
  },
  issues: [
    {
      index: 0,
      id: '6292',
      text: 'Other - stuff and things',
      benefitType: 'vha',
      date: null,
      beforeAma: true,
      ineligibleReason: null,
      vacolsId: null,
      vacolsSequenceId: null,
      vacolsIssue: null,
      decisionReviewTitle: 'Higher-Level Review',
      withdrawalDate: null,
      endProductCleared: null,
      endProductCode: null,
      category: 'Other',
      editable: true,
      examRequested: null,
      decisionIssueId: null,
      isPreDocketNeeded: null
    },
    {
      index: 1,
      text: 'Beneficiary Travel - vha camo testing',
      benefitType: 'vha',
      date: '2023-07-19',
      timely: true,
      beforeAma: false,
      ineligibleReason: null,
      decisionReviewTitle: null,
      category: 'Beneficiary Travel',
      editable: true,
      isPreDocketNeeded: null
    }
  ],
  featureToggles: {
    useAmaActivationDate: true,
    correctClaimReviews: true,
    covidTimelinessExemption: true
  },
  formType: 'higher_level_review',
  userCanWithdrawIssues: false,
  withdrawReview: false
};
