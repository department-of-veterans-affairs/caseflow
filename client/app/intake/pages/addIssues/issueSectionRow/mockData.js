/* eslint-disable no-empty-function */
const issueSectionRowProps = {
  editPage: true,
  featureToggles: {},
  formType: 'test',
  intakeData: {},
  onClickIssueAction: () => {},
  sectionIssues: [
    {
      index: 0,
      id: '4277',
      text: 'Medical and Dental Care Reimbursement - Issue',
      benefitType: 'vha',
      date: '2023-07-11',
      decisionReviewTitle: 'Supplemental Claim',
      category: 'Medical and Dental Care Reimbursement',
      editable: true,
    },
  ],
  userCanWithdrawIssues: true,
  withdrawReview: false
};

export { issueSectionRowProps };
