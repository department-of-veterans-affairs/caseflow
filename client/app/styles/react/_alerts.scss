
.usa-alert {

  &.fixed {
    position: fixed;
    top: 0;
    margin-top: 0;
    z-index: 999;
    left: 50%;
    transform: translate(-50%, 0);
  }

  // Remove after USWDS V.2.0
  &.usa-alert-slim {
    background-size: 4rem;

    .usa-alert-body {
      padding-left: 3.5rem;
    }

    .usa-alert-text {
      font-size: 2rem;
    }
  }

  .usa-alert-text {
    line-height: 1.5;
  }
}
