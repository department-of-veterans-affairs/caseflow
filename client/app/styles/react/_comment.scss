.comment-control-button-container {
  display: flex;
  width: 100%;
  padding: 3px 0;
  justify-content: space-between;
}

.comment-save-button-container {
  display: inline-block;
  width: 100%;
  margin-top: 5px;
  margin-bottom: 17px;
}

.comment-control-button {
  padding: 0;

  &.horizontal {
    text-align: left;
  }
}

.comment-control-button-divider {
  padding: 0 10px;
}

.comment-container-selected {
  box-shadow: 3px 3px 5px $color-gray-light;
}

.comment-container {
  word-wrap: break-word;
  word-break: break-word;
  background-color: $color-white;
  padding: 15px;
  cursor: pointer;

  &:not(.comment-horizontal-container) {
    margin-bottom: 15px;
    border-radius: 3px;
    border: 1px solid $color-gray-lighter;
  }

  &.comment-horizontal-container {
    padding: 25px 25px 25px 40px;
  }
}

// Max width is enforced so the size of this comment box
// fits with existing textareas in style guide
.comment-size-container {
  max-width: 430px;

  .cf-form-textarea {
    margin-bottom: 20px;
  }
}

.comment-horizontal-container {
  display: inline-block;
  width: 100%;
}

.comment-horizontal-spacing {
  margin-right: 49px;
}

.commentIcon-container {
  position: absolute;
  z-index: 10;
  // These dimensions need to match the comment icon SVG dimensions.
  width: 40px;
  height: 40px;
}

