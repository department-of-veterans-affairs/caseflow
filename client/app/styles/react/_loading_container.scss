.loadingContainer-positioning {
  position: absolute;
  width: 100%;
  height: 100%;
}

.loadingContainer-table {
  width: 100%;
  height: 100%;
  display: table;
}

.loadingContainer-table-cell {
  text-align: center;
  display: table-cell;
  vertical-align: middle;
}

.loadingContainer-content {
  position: relative;
  z-index: 1;

  iframe {
    width: 100%;
    height: 80vh;

    &.form9Viewer {
      height: 500px;
    }
  }
}

.loadingContainer-container {
  position: relative;
  overflow: hidden;
}

.cf-loading-button-text {
  display: inline-block;
  margin-right: 10px;
  line-height: 0;
  position: relative;
}

.cf-loading-message {
  .loading-message-text {
    display: inline-block;
    font-style: italic;
    font-weight: bolder;
  }

}

.cf-loading-button-symbol {
  display: inline-block;
}
