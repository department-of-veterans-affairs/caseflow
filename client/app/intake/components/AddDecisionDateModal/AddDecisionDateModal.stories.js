import React from 'react';
import ReduxBase from 'app/components/ReduxBase';
import { reducer, generateInitialState } from 'app/intake';
import AddDecisionDateModal from './AddDecisionDateModal';
import mockData from './mockData';

const ReduxDecorator = (Story) => (
  <ReduxBase reducer={reducer} initialState={generateInitialState()}>
    <Story />
  </ReduxBase>
);

export default {
  component: AddDecisionDateModal,
  decorators: [ReduxDecorator],
  title: 'Intake/Edit Issues/Add Decision Date Modal',
};

export const Basic = () => {
  const { closeHandler, currentIssue, index } = mockData;

  return (
    <AddDecisionDateModal closeHandler={closeHandler} currentIssue={currentIssue} index={index} />
  );
};
