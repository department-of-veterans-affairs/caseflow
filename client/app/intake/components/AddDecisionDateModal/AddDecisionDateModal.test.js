import React from 'react';
import { render, screen } from '@testing-library/react';
import AddDecisionDateModal from './AddDecisionDateModal';
import mockData from './mockData';

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: () => jest.fn().mockImplementation(() => Promise.resolve(true)),
}));

// Ensures the snapshot always matches the same date.
const fakeDate = new Date(2023, 7, 10, 0, 0, 0, 0);

describe('AddDecisionDateModal', () => {

  beforeAll(() => {
    // Ensure consistent handling of dates across tests
    jest.useFakeTimers('modern');
    jest.setSystemTime(fakeDate);
  });

  afterAll(() => {
    jest.clearAllMocks();
    jest.useRealTimers();
  });

  const setup = (testProps) =>
    render(
      <AddDecisionDateModal
        {...testProps}
      />
    );

  it('renders', () => {
    const modal = setup(mockData);

    expect(modal).toMatchSnapshot();
    expect(screen.getByText('Add Decision Date')).toBeInTheDocument();
  });

  it('displays Edit Decision Date if the issue has an editedDecisionDate', () => {
    setup({ ...mockData, currentIssue: { ...mockData.currentIssue, editedDecisionDate: '12/7/2017' } });

    expect(screen.getByText('Edit Decision Date')).toBeInTheDocument();
  });

  it('disables save button if no date is present', () => {
    setup(mockData);
    const save = screen.getByText('Save');

    expect(save).toHaveAttribute('disabled');
  });
});
