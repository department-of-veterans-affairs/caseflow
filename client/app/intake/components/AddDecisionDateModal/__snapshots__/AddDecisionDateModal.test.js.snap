// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddDecisionDateModal renders 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <section
          aria-describedby="modal_id-desc"
          aria-labelledby="modal_id-title"
          aria-modal="true"
          class="cf-modal active "
          id="modal_id"
          role="dialog"
        >
          <div
            class="cf-modal-body"
            id=""
          >
            <button
              class="cf-modal-close"
              id="Add-Decision-Date-button-id-close"
              type="button"
            >
              <span
                class="usa-sr-only"
              >
                Close
              </span>
              <svg
                class="cf-icon-close"
                height="55"
                viewBox="0 0 55 55"
                xmlns="http://www.w3.org/2000/svg"
              >
                <title>
                  close
                </title>
                <g
                  fill="#212121"
                >
                  <path
                    d="M52.6 46.9l-6 6c-.8.8-1.9 1.2-3 1.2s-2.2-.4-3-1.2l-13-13-13 13c-.8.8-1.9 1.2-3 1.2s-2.2-.4-3-1.2l-6-6c-.8-.8-1.2-1.9-1.2-3s.4-2.2 1.2-3l13-13-13-13c-.8-.8-1.2-1.9-1.2-3s.4-2.2 1.2-3l6-6c.8-.8 1.9-1.2 3-1.2s2.2.4 3 1.2l13 13 13-13c.8-.8 1.9-1.2 3-1.2s2.2.4 3 1.2l6 6c.8.8 1.2 1.9 1.2 3s-.4 2.2-1.2 3l-13 13 13 13c.8.8 1.2 1.9 1.2 3s-.4 2.2-1.2 3z"
                  />
                </g>
              </svg>
            </button>
            <div
              style="display: flex;"
            >
              <div
                data-css-1gs0ko2=""
              >
                <h1
                  id="modal_id-title"
                >
                  Add Decision Date
                </h1>
                <div
                  data-css-o98ubo=""
                >
                  <div>
                    <strong
                      data-css-1qn6dtd=""
                    >
                      Issue:
                    </strong>
                    Beneficiary Travel
                  </div>
                  <div>
                    <strong
                      data-css-1qn6dtd=""
                    >
                      Benefit type:
                    </strong>
                    Veterans Health Administration
                  </div>
                  <div>
                    <strong
                      data-css-1qn6dtd=""
                    >
                      Issue description:
                    </strong>
                    Beneficiary Travel - Issue Description
                  </div>
                  <div
                    data-css-xsot8h=""
                  >
                    <div
                      class="cf-form-textinput   "
                    >
                      <label
                        for="decision-date"
                      >
                        <span>
                          Decision date
                        </span>
                      </label>
                      <input
                        class="cf-form-textinput"
                        id="decision-date"
                        max="2023-08-10"
                        name="decision-date"
                        placeholder="mm/dd/yyyy"
                        type="date"
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="cf-modal-divider"
            />
            <div
              class="cf-modal-controls"
            >
              <span>
                <button
                  class="cf-modal-link cf-btn-link close-modal cf-push-left usa-button"
                  id="Add-Decision-Date-button-id-0"
                  type="button"
                >
                  Cancel
                </button>
              </span>
              <span>
                <button
                  class="save-issue cf-push-right usa-button-disabled usa-button"
                  disabled=""
                  id="Add-Decision-Date-button-id-1"
                  type="button"
                >
                  Save
                </button>
              </span>
            </div>
          </div>
        </section>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <section
        aria-describedby="modal_id-desc"
        aria-labelledby="modal_id-title"
        aria-modal="true"
        class="cf-modal active "
        id="modal_id"
        role="dialog"
      >
        <div
          class="cf-modal-body"
          id=""
        >
          <button
            class="cf-modal-close"
            id="Add-Decision-Date-button-id-close"
            type="button"
          >
            <span
              class="usa-sr-only"
            >
              Close
            </span>
            <svg
              class="cf-icon-close"
              height="55"
              viewBox="0 0 55 55"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                close
              </title>
              <g
                fill="#212121"
              >
                <path
                  d="M52.6 46.9l-6 6c-.8.8-1.9 1.2-3 1.2s-2.2-.4-3-1.2l-13-13-13 13c-.8.8-1.9 1.2-3 1.2s-2.2-.4-3-1.2l-6-6c-.8-.8-1.2-1.9-1.2-3s.4-2.2 1.2-3l13-13-13-13c-.8-.8-1.2-1.9-1.2-3s.4-2.2 1.2-3l6-6c.8-.8 1.9-1.2 3-1.2s2.2.4 3 1.2l13 13 13-13c.8-.8 1.9-1.2 3-1.2s2.2.4 3 1.2l6 6c.8.8 1.2 1.9 1.2 3s-.4 2.2-1.2 3l-13 13 13 13c.8.8 1.2 1.9 1.2 3s-.4 2.2-1.2 3z"
                />
              </g>
            </svg>
          </button>
          <div
            style="display: flex;"
          >
            <div
              data-css-1gs0ko2=""
            >
              <h1
                id="modal_id-title"
              >
                Add Decision Date
              </h1>
              <div
                data-css-o98ubo=""
              >
                <div>
                  <strong
                    data-css-1qn6dtd=""
                  >
                    Issue:
                  </strong>
                  Beneficiary Travel
                </div>
                <div>
                  <strong
                    data-css-1qn6dtd=""
                  >
                    Benefit type:
                  </strong>
                  Veterans Health Administration
                </div>
                <div>
                  <strong
                    data-css-1qn6dtd=""
                  >
                    Issue description:
                  </strong>
                  Beneficiary Travel - Issue Description
                </div>
                <div
                  data-css-xsot8h=""
                >
                  <div
                    class="cf-form-textinput   "
                  >
                    <label
                      for="decision-date"
                    >
                      <span>
                        Decision date
                      </span>
                    </label>
                    <input
                      class="cf-form-textinput"
                      id="decision-date"
                      max="2023-08-10"
                      name="decision-date"
                      placeholder="mm/dd/yyyy"
                      type="date"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="cf-modal-divider"
          />
          <div
            class="cf-modal-controls"
          >
            <span>
              <button
                class="cf-modal-link cf-btn-link close-modal cf-push-left usa-button"
                id="Add-Decision-Date-button-id-0"
                type="button"
              >
                Cancel
              </button>
            </span>
            <span>
              <button
                class="save-issue cf-push-right usa-button-disabled usa-button"
                disabled=""
                id="Add-Decision-Date-button-id-1"
                type="button"
              >
                Save
              </button>
            </span>
          </div>
        </div>
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
